control-plane/bfruntime.cpp

```diff
@@ -866,6 +866,67 @@ BFRuntimeGenerator::addMeters(Util::JsonArray* tablesJson) const {
     }
 }
 
+Util::JsonObject*
+BFRuntimeGenerator::makeTypeBitsOrBytes(int bitwidth,
+                                             boost::optional<cstring> defaultValue) const {
+    auto* type_obj = new Util::JsonObject();
+    if (bitwidth % 8) {
+        type_obj->emplace("type", "bits");
+        type_obj->emplace("width", bitwidth);
+    } else {
+        type_obj->emplace("type", "bytes");
+        type_obj->emplace("width", bitwidth >> 3);
+    }
+    if (defaultValue != boost::none)
+        type_obj->emplace("default_value", *defaultValue);
+    return type_obj;
+}
+
+void
+BFRuntimeGenerator::addMatchValueLookupTables(Util::JsonArray* tables_json_array) const {
+    for (const auto& emvlt : p4info.emvlts()) {
+         const auto pre = emvlt.preamble();
+         auto* annotations = transformAnnotations(pre.annotations().begin(),
+                                                  pre.annotations().end());
+         cstring table_name = IR::P4Program::main + "." + pre.name();
```

Comment:
Is this some standard ? What does it give as output table_name

Reply:
i'm not sure but it's like this "MainControlImpl.mvlut_object"